watson/utils.py

```diff
@@ -197,3 +197,12 @@ def safe_save(path, content, ext='.bak'):
             os.rename(path, path + ext)
 
         os.rename(tmpfp.name, path)
+
+
+def deduplicate(sequence):
+    """
+    Return the input sequence without duplicate, keep the order.
```

Comment:
More accurately:

    Return a list with all items of the input sequence but duplicates removed.
    
    Keeps the order of items in the input sequence.